main/com.tunein/manifest.tt

```diff
@@ -0,0 +1,111 @@
+// FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+// COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+// INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+// OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Author: Jake Wu <jmhw0123@gmail.com>
+
+class @com.tunein
+#_[thingpedia_name="Tunein"]
+#_[thingpedia_description="Listen to internet radio"]
+#_[canonical="radio"]
+#[license="BSD-3-Clause"]
+#[license_gplcompatible=true]
+#[subcategory="media"]
+{
+  import loader from @org.thingpedia.v2();
+  import config from @org.thingpedia.config.none();
+  
+  entity station #_[description="Radio stations on TuneIn"];
+  entity show #_[description="Show names from the station on TuneIn"];
+
+  list query station(out id: Entity(com.tunein:station)
+                     #[unique=true]
+                     #[string_values="com.tunein:station"]
+                     #_[canonical={
+                       default="base",
+                       base=["name", "title"],
+                       property=["name #", "title #", "channel #"],
+                       passive_verb=["named #", "called #", "titled #"],
+                     }],
+                     out show: Entity(com.tunein:show)
+                     #[unique=true]
+                     #[string_values="com.tunein:show"]
+                     #_[canonical={
+                       default="base",
+                       base=["name", "title", "track", "show", "program", "current show", 
+                             "current program", "now playing", "currently playing"],
+                       passive_verb=["named #", "called #", "titled #", "related to #"],
+                       property=["name #", "title #", "now playing #", "currently playing #"]
+                     }],
+                     out link: Entity(tt:url)
+                     #[filterable=false]
+                     #_[canonical=["link", "url"]],
+                     out image: Entity(tt:picture)
+                     #[filterable=false]
+                     #_[canonical={
+                       default="base",
+                       base=["picture", "image", "photo", "cover"]
+                     }]
+  )
+  #_[canonical=["radio channel", "radio show", "radio station"]]
+  #_[canonical_short=["radio"]]
+  #_[on_error={
+    query_invalid=["radio station does not exist"],
+    bad_request=["malformed radio station name"],
+    unauthorized=["radio service is not accessible without a valid user name and password"],
+    service_unavailable=["radio service is temporarily not available"],
+    network_error=["network_error"]
```

Comment:
Let network errors bubble up, the general Genie code should handle those correctly already.